data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3SelectObjectWorker.java

```diff
@@ -92,16 +97,27 @@ public S3SelectObjectWorker(final S3ObjectRequest s3ObjectRequest) {
      *
      * @throws IOException When Input/Output exception occur while processing the data.
      */
-    public void parseS3Object(final S3ObjectReference s3ObjectReference) throws IOException {
+    public void parseS3Object(final S3ObjectReference s3ObjectReference, final AcknowledgementSet acknowledgementSet) throws IOException {
         try{
-            selectObjectContent(s3ObjectReference);
+            selectObjectContent(s3ObjectReference, acknowledgementSet);
         }catch(Exception e){
             s3ObjectPluginMetrics.getS3ObjectsFailedCounter().increment();
             throw new IOException(e);
         }
     }
 
-    private void selectObjectContent(S3ObjectReference s3ObjectReference) throws IOException {
+    /**
+     * Fetch the S3 object content using s3 select service and parsing and pushing to buffer.
+     * @param s3ObjectReference Contains bucket and s3 object details
+     * @param acknowledgementSet acknowledgement set for the object
+     *
+     * @throws IOException exception is thrown every time because this is not supported
+     */
```

Comment:
I'm not clear why this Javadoc is added here. `acknowledgementSet` is not a parameter in below method.

Reply:
Removed javadoc here. For some reason it was there initially in this file. I tried to keep it that way. But since I added it to the interface file, I can remove it here.